include/alpaka/SimdWhereExpr.hpp

```diff
@@ -0,0 +1,79 @@
+/* Copyright 2025 René Widera
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+/** @file This file provides a basic implementation of a SIMD vector.
+ *
+ * The implementation is based on the class Vec:
+ *   - the storge policy should become the native SIMD implementation e.g. std::simd
+ *   - load/ store and simd specifis should be implemented in the storage policy
+ *   - the name of storage policy should be changed
+ *
+ *   The current operator operations relay on compilers auto vectorization.
+ */
+
+#pragma once
+
+#include "alpaka/Simd.hpp"
+
+namespace alpaka
+{
+    template<concepts::SimdMask Mask, concepts::Simd T_Simd>
+    struct SimdWhereExpr
+    {
+        Mask const& mask;
+        T_Simd& value;
+
+        constexpr SimdWhereExpr(Mask const& m, T_Simd& v) : mask(m), value(v)
+        {
+        }
+
+        // disable copy and move constructors/operators to avoid pointing to invalid references.
+        constexpr SimdWhereExpr(SimdWhereExpr const&) = delete;
+        constexpr SimdWhereExpr(SimdWhereExpr&&) = delete;
+        constexpr SimdWhereExpr& operator=(SimdWhereExpr const&) = default;
+        constexpr SimdWhereExpr& operator=(SimdWhereExpr&&) = default;
```

Comment:
Only the constructors are disable. The copy and move assignment are enable. Is the code or comment wrong?